dashboard/test/helpers/delete_accounts_helper_test.rb

```diff
@@ -273,6 +273,21 @@ class DeleteAccountsHelperTest < ActionView::TestCase
       'Expected no activity record that references a level source to exist for this user'
   end
 
+  #
+  # Table: dashboard.user_levels
+  #
+
+  test "Disconnects user_levels from level_sources" do
+    user_level = create :user_level, level_source: create(:level_source)
+
+    refute_nil user_level.level_source_id
+
+    purge_user user_level.user
+    user_level.reload
+
+    assert_nil user_level.level_source_id
```

Comment:
it looks like the level_source could become an orphan here, but I don't see a quick solution for that since the level_source_id column is not indexed.